pages/instructorAssessmentStatistics/instructorAssessmentStatistics.js

```diff
@@ -31,24 +32,23 @@ router.get('/', function (req, res, next) {
   setFilenames(res.locals);
   async.series(
     [
-      function (callback) {
-        debug('query assessment_stats');
-        var params = { assessment_id: res.locals.assessment.id };
-        sqldb.queryOneRow(sql.assessment_stats, params, function (err, result) {
-          if (ERR(err, callback)) return;
-          res.locals.assessment_stat = result.rows[0];
-          callback(null);
+      async () => {
+        // make sure statistics are up to date
+        await assessment.updateAssessmentStatistics(res.locals.assessment.id);
+      },
+      async () => {
+        // re-fetch assessment to get updated statistics
+        const result = await sqldb.queryOneRowAsync(sql.select_assessment, {
+          assessment_id: res.locals.assessment.id,
         });
+        res.locals.assessment = result.rows[0].assessment;
       },
-      function (callback) {
-        debug('query assessment_duration_stats');
-        // FIXME: change to assessment_instance_duration_stats and show all instances
-        var params = { assessment_id: res.locals.assessment.id };
-        sqldb.queryOneRow(sql.assessment_duration_stats, params, function (err, result) {
-          if (ERR(err, callback)) return;
-          res.locals.duration_stat = result.rows[0];
-          callback(null);
+      async () => {
+        // get formatted duration statistics
+        const result = await sqldb.queryOneRowAsync(sql.select_duration_stats, {
+          assessment_id: res.locals.assessment.id,
```

Comment:
Any reason why these queries are separate? They could very easily be merged.

Reply:
They are needed separately in the `GET(:filename)` handler (see a83518ec7a0ab8627258a3e39e5e288856223925).